components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java

```diff
@@ -110,8 +156,35 @@ public void testGetScopesWithStartAndCount(Integer startIndex, Integer count) th
         assertNotNull(oAuth2ScopeService.getScopes(startIndex, count), "Expected a not null object");
     }
 
+    @Test
+    public void testGetScopesWithStartAndCountAndRequestedScopes() throws Exception {
+
+        assertNotNull(oAuth2ScopeService.getScopes(1, 2, false, "true"), "Expected a not null object");
+    }
+
+    @Test()
+    public void testIsScopeExistsWithNullName() throws IdentityOAuth2ScopeException {
+
+        try {
+            oAuth2ScopeService.isScopeExists(null);
```

Comment:
Test should fail if registerScope() doesn't throw an exception

Check other methods which assert for exceptions